src/status_im/contexts/wallet/sheets/account_origin/view.cljs

```diff
@@ -41,5 +42,5 @@
      :size            24
      :icon-left       :i/info
      :container-style style/action-container
-     :on-press        #(rn/open-url const/create-account-link)}
+     :on-press        #(rf/dispatch [:open-url const/create-account-link])}
```

Comment:
I don't see any advantage or need on wrapping a single effect call in a re-frame event to just call with dispatch. 
Using a regular function call is more straightforward and not a bad practice.

@seanstrom Is there any reason behind doing this? :thinking: 
CC: @ilmotta wdyt?

Reply:
I prefer what @seanstrom did because it leads to a more consistent and simpler UI layer long-term.

The view shouldn't know or worry if certain events are simple wrappers around effects or more complicated ones. To the view layer, all it should know is to dispatch events. Of course there are exceptions to any rule.

----

Going a bit deeper, the case for consistency to me is that it's hard to draw the line of what side-effects are okay calling in the UI and which are not.

I've seen many times in this repo that just because side-effects are put inside callbacks, the dev would think it's okay to write highly mutable and side-effectful code. I did this a few times myself, not proud.

The code below is the kind of code that ends up existing in a code base where side effects are directly called in the view layer. I think we would write more maintainable code in general if we were more strongly against solutions like this. `rn/open-url` is harmless I know, not the best example, but it very slightly encourages the wrong practice from my perspective.

https://github.com/status-im/status-mobile/blob/babcd96fb625aa843d373ced160d94fd6e54bbd9/src/status_im/contexts/wallet/add_account/create_account/import_private_key/view.cljs#L19-L34

Reply:
> I don't see any advantage or need on wrapping a single effect call in a re-frame event to just call with dispatch. Using a regular function call is more straightforward and not a bad practice.
> 
> @seanstrom Is there any reason behind doing this?

For context, this change was discussed previously here: https://github.com/status-im/status-mobile/pull/20063#discussion_r1619194373

In my opinion, the main reason to call `rf/dispatch` vs the `rn/open-url` function is to eventually standardise on the effects that we support in the app. So when someone is creating some UI, they can choose from a library of supported interactions that follow a convention. We can allow developers to call side-effect functions directly, but we can end up coupling the views to side-effect implementations.

On a slightly different note, I like being able to trace the events and effects that are happening inside the app with Re-Frisk, it's been very helpful when debugging stuff.

Reply:
Thanks for sharing your perspectives @ilmotta @seanstrom 

Now I agree :+1: